src/state.cairo

```diff
@@ -13,23 +19,21 @@ pub struct ChainState {
     pub epoch_start_time: u32,
     /// Previous timestamps.
     pub prev_timestamps: Span<u32>,
-    // TODO: utreexo_roots?
+    // Utreexo roots (for checking [TxIn] inclusion proofs)
+    pub utreexo_roots: Span<felt252>
```

Comment:
Don't we want to have a separate accumulator type here? And not sure if felt252 is adequate, just starting to read the spec.

Reply:
Yes, it makes sense to introduce a separate type (though I'm not sure if it would contain more than the roots). We will probably use a modified variant of Utreexo that uses zk friendly hash function hence the felt type.